### DISCUSSION TITLE
Extending the Query Loop default fields

### DISCUSSION THREAD

**Discussion by paaljoachim** [WordPress/developer-blog-content#324] [Topic Ideas]
Currently the Query Loop will show the default fields of Title, Featured image, date and excerpt.
It would be helpful to extend the Query Loop to show additional fields.

A use case.
An artist has an album post. It uses the standard fields. 
Title = album name
Featured image = album cover
Excerpt = a description of the album
Date = date when the album came out

He would like to add artists whom contributed to the album.
He especially would like to add album tracks.

This means two additional fields:
Contributed to album 
and 
tracks

As I see it one would need to define specific blocks that will be seen in the Album preview page where the Query loop would also show these specific blocks.

It would actually be very helpful to have two methods for this. Likely the regular Json file and code snippets. Code snippets would be an easy way to add a snippet directly to the web site without even having access to the files outside of WordPress. Having a Code snippet plugin one could just add the snippet turn it on and see the result. Go forth and back between the result and the snippet to easily adjust the code.

**Comment by justintadlock** [comment]
Just some quick and rough thoughts on data structures for the use case...

For artists, there are a few different methods of handling that:

- User data: needs a method of handling post relationships (can be done via meta), assuming the artists have user profiles on the site.
- Custom taxonomy: `artist` taxonomy.
- Post meta: Could be a simple input field and work with block bindings easily.
- A custom block (can be tied to either the post content, meta, or another data source).

For tracks:

- Custom post type: Assuming the tracks should be discoverable on their own as "posts," and you'd want to handle the connection through post relationships.
- Post meta: Might be harder if meant to be presented as a list, at least if wanting to tie to block bindings.
- A custom block (can be tied to either the post content, meta, or another data source).

As for extending the Query Loop block, that's generally fairly straightforward in most cases once you have the other pieces in place.

**Comment by paaljoachim** [comment]
Here it would be great to start with the approach to what is easiest to accomplish.
Then gradually expand from the simple to the intermediate and then the advanced level.
How one can use tools that exist in WordPress today or will be coming soon.

For instance the artist used an AI tool and have now installed CPT UI and Advanced Custom Fields and has been exploring those tools.
(These tools do not require access to the files outside of the WordPress backend.) 
It would be nice to do the same thing with native WordPress core tools if possible.

**Comment by bph** [comment]
Topic approved, conversation continues on the issue
- #333
